lib/classes/TemporalCollection.js

```diff
@@ -90,12 +90,46 @@ export default class TemporalCollection extends PromisedMongoCollection {
     return result.value;
   }
 
-  aggregate(pipeline={}, options) {
-    return this.aggregateRaw({ ...pipeline, $match: addCurrent(pipeline.$match) }, options);
+
+  /**
+   * Add restrictions to aggregation pipeline for temporal docs
+   *
+   * @param {Function} restriction : function to modify initial $match
+   * @param {Date} date : date for temporal restriction
+   * @param {Array} pipeline : aggregation pipeline
+   * @param {Object} options : aggregation options
+   * @return {Promise}
+   */
+  _aggregateWrapper(restriction, date, pipeline) {
+    if (!Array.isArray(pipeline)) {
+      throw new Error(`Temporal Mongo: pipeline must be an array!`);
+    }
+
+    if (!pipeline.length) {
+      return this.aggregateRaw();
+    }
+
+    // first element of pipeline should be $match to current
+    const [ first ] = pipeline;
+
+    if (first.$match) {
+      first.$match = restriction(first.$match, date);
+    } else {
+      pipeline.unshift({ $match: restriction({}, date) });
+    }
+
+    return this.aggregateRaw(...pipeline);
   }
 
-  aggregateByDate(pipeline={}, options, date) {
-    return this.aggregateRaw({ ...pipeline, $match: addDate(pipeline.$match, date) }, options);
+  aggregate(...pipeline) {
+    return this._aggregateWrapper(addCurrent, null, pipeline);
```

Comment:
I'm a little confused by the spread operators. Shouldn't the first argument to `aggregate` be an array already?

While it's true that `promised-mongo` passes an object in the example you linked (and in their [`aggregate` test](https://github.com/gordonmleigh/promised-mongo/blob/master/test/Collection.js#L28), I think this is actually a bug and should be passing an array (which is the Mongo API):
https://github.com/gordonmleigh/promised-mongo/issues/27

Our app code uses that array format too.